src/interface/paint_path.c

```diff
@@ -0,0 +1,775 @@
+    uint8 bl = direction | diagonal;
+
+    if (bp & 2) {
+        // Bottom right of tile is a tunnel
+        registers regs = {0};
+        regs.eax = 0xFFFFFFFF;
+        if (bl == 5) {
+            regs.ax = (height + 16) / 16;
+            regs.ah = 0x0A;
+        } else if (bp & 1) {
+            regs.ax = height / 16;
+            regs.ah = 0x0B;
+        } else {
+            regs.ax = height / 16;
+            regs.ah = 0x0A;
+
+        }
+        RCT2_ADDRESS(0x009E30B6, uint32)[RCT2_GLOBAL(0x141F56B, uint8) / 2] = regs.eax;
+        RCT2_GLOBAL(0x141F56B, uint8)++;
+    }
+
+    if (!(bp & 4)) {
+        return;
+    }
+
+    // Bottom left of the tile is a tunnel
+    registers regs = {0};
+    regs.eax = 0xFFFFFFFF;
+    if (bl == 6) {
+        regs.ax = (height + 16) / 16;
+        regs.ah = 0x0A;
+    } else if (bp & 8) {
+        regs.ax = height / 16;
+        regs.ah = 0x0B;
+    } else {
+        regs.ax = height / 16;
+        regs.ah = 0x0A;
+
+    }
+    RCT2_ADDRESS(0x009E3138, uint32)[RCT2_GLOBAL(0x141F56A, uint8) / 2] = regs.eax;
+    RCT2_GLOBAL(0x141F56A, uint8)++;
+}
+
+void viewport_path_paint_setup(uint8 direction, int height, rct_map_element * map_element)
+{
+    RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_FOOTPATH;
+
+    uint32 ecx = get_current_rotation();
+    bool word_F3F038 = false;
+
+    uint32 dword_F3EF74 = 0;
+    uint32 dword_F3EF70 = 0;
+
+    if (RCT2_GLOBAL(0x9DEA6F, uint8) & 1) {
+        if (map_element->type & 1) {
+            if (map_element->properties.path.ride_index != RCT2_GLOBAL(0x00F64DE8, uint8)) {
+                return;
+            }
+        }
+
+        if (!is_map_element_saved(map_element)) {
+            dword_F3EF70 = 0x21700000;
+        }
+    }
+
+    if (footpath_element_path_scenery_is_ghost(map_element)) {
+        dword_F3EF74 = RCT2_ADDRESS(0x993CC4, uint32_t)[gConfigGeneral.construction_marker_colour];
+    }
+
+    if (map_element->flags & MAP_ELEMENT_FLAG_GHOST) {
+        RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_NONE;
+        dword_F3EF70 = RCT2_ADDRESS(0x993CC4, uint32_t)[gConfigGeneral.construction_marker_colour];
+    }
+
+    sint16 x = RCT2_GLOBAL(0x009DE56A, sint16), y = RCT2_GLOBAL(0x009DE56E, sint16);
+
+    rct_map_element * surface = map_get_surface_element_at(x / 32, y / 32);
+
+    uint16 bl = height / 8;
+    if (surface == NULL) {
+        word_F3F038 = true;
+    } else if (surface->base_height != bl) {
+        word_F3F038 = true;
+    } else {
+        if (map_element->properties.path.type & 0x04) {
+            // Diagonal path
+
+            if ((surface->properties.surface.slope & 0x1F) != byte_98D800[map_element->properties.path.type & 0x03]) {
+                word_F3F038 = true;
+            }
+        } else {
+            if (surface->properties.surface.slope & 0x1F) {
+                word_F3F038 = true;
```

Comment:
it seems like `word_F3F038` is the flag stating that you should draw flooring underneath the tile i.e. path in the air/slopped.